@Karlatemp 

我2.14.0能稳定复现
复现: 还在找
日志:
```
2023-02-14 12:17:08 W/Bot 1403035170180600110: An exception occurred when processing event. Subscriber scope: '<unnamed>'. Broadcaster scope: 'EventDispatcher'
kotlinx.serialization.SerializationException: Class 'MockImage' is not registered for polymorphic serialization in the scope of 'SingleMessage'.
Mark the base class as 'sealed' or register the serializer explicitly.
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:102)
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:113)
	at kotlinx.serialization.PolymorphicSerializerKt.findPolymorphicSerializer(PolymorphicSerializer.kt:109)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at kotlinx.serialization.internal.CollectionLikeSerializer.serialize(CollectionSerializers.kt:69)
	at net.mamoe.mirai.message.data.MessageChain$Serializer.serialize(MessageChain.kt:277)
	at net.mamoe.mirai.message.data.MessageChain$Serializer.serialize(MessageChain.kt:272)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:219)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData.write$Self(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData$$serializer.serialize(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData$$serializer.serialize(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$special$$inlined$map$1.serialize(Serialization.kt:47)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl.serialize(MessageSerializersImpl.kt)
	at net.mamoe.mirai.internal.message.source.OfflineMessageSourceImplData$Serializer.serialize(offlineSourceImpl.kt)
	at net.mamoe.mirai.internal.message.source.OfflineMessageSourceImplData$Serializer.serialize(offlineSourceImpl.kt:46)
	at net.mamoe.mirai.internal.message.protocol.impl.QuoteReplyProtocol$collectProcessorsImpl$lambda-3$$inlined$map$1.serialize(Serialization.kt:47)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:219)
	at kotlinx.serialization.json.Json.encodeToString(Json.kt:85)
	at tech.eritquearcus.miraicp.shared.Packets$Outgoing$MessageEventData.<init>(Packet.kt:576)
	at tech.eritquearcus.miraicp.shared.Packets$Utils.toEventData(Packet.kt:789)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invokeSuspend(PublicShared.kt:840)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invoke(PublicShared.kt)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invoke(PublicShared.kt)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invokeSuspend(EventChannel.kt:480)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invokeSuspend(EventChannel.kt:524)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invokeSuspend(EventChannel.kt:524)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invokeSuspend(SafeListener.kt:58)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invoke(SafeListener.kt)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invoke(SafeListener.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at net.mamoe.mirai.internal.event.SafeListener.onEvent(SafeListener.kt:58)
	at net.mamoe.mirai.internal.event.EventListeners.process(EventListeners.kt:120)
	at net.mamoe.mirai.internal.event.EventListeners.callListeners$mirai_core(EventListeners.kt:68)
	at net.mamoe.mirai.internal.event.EventChannelImpl.callListeners(EventChannelImpl.kt:85)
	at net.mamoe.mirai.internal.event.EventChannelImpl.broadcastEventImpl(EventChannelImpl.kt:75)
	at net.mamoe.mirai.internal.network.components.EventDispatcherImpl.broadcast$suspendImpl(EventDispatcher.kt:99)
	at net.mamoe.mirai.internal.network.components.EventDispatcherImpl.broadcast(EventDispatcher.kt)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.access$broadcast$s134264449(MockEventDispatcherImpl.kt:29)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invokeSuspend(MockEventDispatcherImpl.kt:41)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invoke(MockEventDispatcherImpl.kt)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invoke(MockEventDispatcherImpl.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:55)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:112)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.broadcast$suspendImpl(MockEventDispatcherImpl.kt:38)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.broadcast(MockEventDispatcherImpl.kt)
	at net.mamoe.mirai.mock.internal.MockMiraiImpl.broadcastEvent(MockMiraiImpl.kt:356)
	at net.mamoe.mirai.event.EventKt.broadcast(Event.kt:157)
	at net.mamoe.mirai.mock.internal.contact.MockNormalMemberImpl.says(MockNormalMemberImpl.kt:206)
	at tech.eritquearcus.miraicp.shared.test.events.OperationTest$imgUploaded$1.invokeSuspend(OperationTest.kt:237)
	(Coroutine boundary)
	at net.mamoe.mirai.internal.event.SafeListener.onEvent(SafeListener.kt:58)
	at net.mamoe.mirai.internal.event.EventListeners.process(EventListeners.kt:120)
	at net.mamoe.mirai.internal.event.EventListeners.callListeners$mirai_core(EventListeners.kt:68)
	at net.mamoe.mirai.internal.event.EventChannelImpl.callListeners(EventChannelImpl.kt:85)
	at net.mamoe.mirai.internal.event.EventChannelImpl.broadcastEventImpl(EventChannelImpl.kt:75)
	at net.mamoe.mirai.internal.network.components.EventDispatcherImpl.broadcast$suspendImpl(EventDispatcher.kt:99)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invokeSuspend(MockEventDispatcherImpl.kt:41)
Caused by: kotlinx.serialization.SerializationException: Class 'MockImage' is not registered for polymorphic serialization in the scope of 'SingleMessage'.
Mark the base class as 'sealed' or register the serializer explicitly.
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:102)
	at kotlinx.serialization.internal.AbstractPolymorphicSerializerKt.throwSubtypeNotRegistered(AbstractPolymorphicSerializer.kt:113)
	at kotlinx.serialization.PolymorphicSerializerKt.findPolymorphicSerializer(PolymorphicSerializer.kt:109)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at kotlinx.serialization.internal.CollectionLikeSerializer.serialize(CollectionSerializers.kt:69)
	at net.mamoe.mirai.message.data.MessageChain$Serializer.serialize(MessageChain.kt:277)
	at net.mamoe.mirai.message.data.MessageChain$Serializer.serialize(MessageChain.kt:272)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:219)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeSerializableElement(AbstractEncoder.kt:80)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData.write$Self(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData$$serializer.serialize(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$SerialData$$serializer.serialize(MessageSerializersImpl.kt:41)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl$special$$inlined$map$1.serialize(Serialization.kt:47)
	at net.mamoe.mirai.internal.message.MessageSourceSerializerImpl.serialize(MessageSerializersImpl.kt)
	at net.mamoe.mirai.internal.message.source.OfflineMessageSourceImplData$Serializer.serialize(offlineSourceImpl.kt)
	at net.mamoe.mirai.internal.message.source.OfflineMessageSourceImplData$Serializer.serialize(offlineSourceImpl.kt:46)
	at net.mamoe.mirai.internal.message.protocol.impl.QuoteReplyProtocol$collectProcessorsImpl$lambda-3$$inlined$map$1.serialize(Serialization.kt:47)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:219)
	at kotlinx.serialization.json.Json.encodeToString(Json.kt:85)
	at tech.eritquearcus.miraicp.shared.Packets$Outgoing$MessageEventData.<init>(Packet.kt:576)
	at tech.eritquearcus.miraicp.shared.Packets$Utils.toEventData(Packet.kt:789)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invokeSuspend(PublicShared.kt:840)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invoke(PublicShared.kt)
	at tech.eritquearcus.miraicp.shared.PublicShared$onEnable$2.invoke(PublicShared.kt)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invokeSuspend(EventChannel.kt:480)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.EventChannel$subscribeAlways$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invokeSuspend(EventChannel.kt:524)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invokeSuspend(EventChannel.kt:524)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.event.FilterEventChannel$intercept$1.invoke(EventChannel.kt)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invokeSuspend(SafeListener.kt:58)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invoke(SafeListener.kt)
	at net.mamoe.mirai.internal.event.SafeListener$onEvent$2.invoke(SafeListener.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:169)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at net.mamoe.mirai.internal.event.SafeListener.onEvent(SafeListener.kt:58)
	at net.mamoe.mirai.internal.event.EventListeners.process(EventListeners.kt:120)
	at net.mamoe.mirai.internal.event.EventListeners.callListeners$mirai_core(EventListeners.kt:68)
	at net.mamoe.mirai.internal.event.EventChannelImpl.callListeners(EventChannelImpl.kt:85)
	at net.mamoe.mirai.internal.event.EventChannelImpl.broadcastEventImpl(EventChannelImpl.kt:75)
	at net.mamoe.mirai.internal.network.components.EventDispatcherImpl.broadcast$suspendImpl(EventDispatcher.kt:99)
	at net.mamoe.mirai.internal.network.components.EventDispatcherImpl.broadcast(EventDispatcher.kt)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.access$broadcast$s134264449(MockEventDispatcherImpl.kt:29)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invokeSuspend(MockEventDispatcherImpl.kt:41)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invoke(MockEventDispatcherImpl.kt)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl$broadcast$2.invoke(MockEventDispatcherImpl.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startCoroutineUndispatched(Undispatched.kt:55)
	at kotlinx.coroutines.CoroutineStart.invoke(CoroutineStart.kt:112)
	at kotlinx.coroutines.AbstractCoroutine.start(AbstractCoroutine.kt:126)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch(Builders.common.kt:56)
	at kotlinx.coroutines.BuildersKt.launch(Unknown Source)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.launch$default(Builders.common.kt:47)
	at kotlinx.coroutines.BuildersKt.launch$default(Unknown Source)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.broadcast$suspendImpl(MockEventDispatcherImpl.kt:38)
	at net.mamoe.mirai.mock.internal.components.MockEventDispatcherImpl.broadcast(MockEventDispatcherImpl.kt)
	at net.mamoe.mirai.mock.internal.MockMiraiImpl.broadcastEvent(MockMiraiImpl.kt:356)
	at net.mamoe.mirai.event.EventKt.broadcast(Event.kt:157)
	at net.mamoe.mirai.mock.internal.contact.MockNormalMemberImpl.says(MockNormalMemberImpl.kt:206)
	at tech.eritquearcus.miraicp.shared.test.events.OperationTest$imgUploaded$1.invokeSuspend(OperationTest.kt:237)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at tech.eritquearcus.miraicp.shared.test.events.OperationTest.imgUploaded(OperationTest.kt:233)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutInvocation.proceed(TimeoutInvocation.java:46)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$7(TestMethodTestDescriptor.java:214)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:135)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:151)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:62)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:113)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:65)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)

Received event: GroupMessageEvent(group=111, senderName=testMember0, sender=222, permission=MEMBER, message=[mirai:source:ids=[-1110530636], internalIds=[-336101144], from group 222 to 111 at 1676395028]imgUploaded[mirai:image:{BCD49145-8622-AD80-04D2-930CEF55E8E1}.png])

```
update: 我的问题，没问题了

--- a/mirai-core-api/src/commonMain/kotlin/contact/announcement/OfflineAnnouncement.kt
+++ b/mirai-core-api/src/commonMain/kotlin/contact/announcement/OfflineAnnouncement.kt
@@ -84,7 +84,7 @@ public sealed interface OfflineAnnouncement : Announcement {
         }
 
         internal object Serializer : KSerializer<OfflineAnnouncement> by OfflineAnnouncementImpl.serializer().map(
-            resultantDescriptor = OfflineAnnouncementImpl.serializer().descriptor.copy(SERIAL_NAME),
+            resultantDescriptor = OfflineAnnouncementImpl.serializer().descriptor,
             deserialize = { it },
             serialize = { it.safeCast<OfflineAnnouncementImpl>() ?: create(content, parameters).cast() }
         )

--- a/mirai-core-api/src/commonMain/kotlin/internal/message/MessageSerializersImpl.kt
+++ b/mirai-core-api/src/commonMain/kotlin/internal/message/MessageSerializersImpl.kt
@@ -38,6 +38,12 @@ public open class MessageSourceSerializerImpl(serialName: String) :
             Mirai.constructMessageSource(botId, kind, fromId, targetId, ids, time, internalIds, originalMessage)
         }
     ) {
+
+    @MiraiInternalApi
+    public companion object {
+        public fun serialDataSerializer(): KSerializer<*> = SerialData.serializer()
+    }
+
     @SerialName(MessageSource.SERIAL_NAME)
     @Serializable
     internal class SerialData(

--- a/mirai-core-api/src/commonMain/kotlin/message/data/FileMessage.kt
+++ b/mirai-core-api/src/commonMain/kotlin/message/data/FileMessage.kt
@@ -25,7 +25,6 @@ import net.mamoe.mirai.message.code.CodableMessage
 import net.mamoe.mirai.message.data.visitor.MessageVisitor
 import net.mamoe.mirai.utils.MiraiInternalApi
 import net.mamoe.mirai.utils.NotStableForInheritance
-import net.mamoe.mirai.utils.copy
 import net.mamoe.mirai.utils.map
 import kotlin.jvm.JvmMultifileClass
 import kotlin.jvm.JvmName
@@ -111,9 +110,9 @@ public expect interface FileMessage : MessageContent, ConstrainSingle, CodableMe
 }
 
 @MiraiInternalApi
-internal open class FallbackFileMessageSerializer constructor(serialName: String) :
+internal open class FallbackFileMessageSerializer :
     KSerializer<FileMessage> by Delegate.serializer().map(
-        Delegate.serializer().descriptor.copy(serialName),
+        Delegate.serializer().descriptor,
         serialize = { Delegate(id, internalId, name, size) },
         deserialize = { Mirai.createFileMessage(id, internalId, name, size) },
     ) {

--- a/mirai-core-api/src/commonMain/kotlin/message/data/RockPaperScissors.kt
+++ b/mirai-core-api/src/commonMain/kotlin/message/data/RockPaperScissors.kt
@@ -102,7 +102,7 @@ public enum class RockPaperScissors(
     }
 
     internal object Serializer : KSerializer<RockPaperScissors> by Surrogate.serializer().map(
-        resultantDescriptor = Surrogate.serializer().descriptor.copy(SERIAL_NAME),
+        resultantDescriptor = Surrogate.serializer().descriptor,
         deserialize = { valueOf(it.name) },
         serialize = { Surrogate(name) },
     ) {

--- a/mirai-core-api/src/commonMain/kotlin/message/data/UnsupportedMessage.kt
+++ b/mirai-core-api/src/commonMain/kotlin/message/data/UnsupportedMessage.kt
@@ -62,7 +62,7 @@ public interface UnsupportedMessage : MessageContent {
     }
 
     public object Serializer : KSerializer<UnsupportedMessage> by Surrogate.serializer().map(
-        resultantDescriptor = Surrogate.serializer().descriptor.copy(SERIAL_NAME),
+        resultantDescriptor = Surrogate.serializer().descriptor,
         deserialize = { Mirai.createUnsupportedMessage(struct.hexToBytes()) },
         serialize = { Surrogate(struct.toUHexString("")) }
     ) {

--- a/mirai-core-api/src/commonMain/kotlin/utils/DeviceInfo.kt
+++ b/mirai-core-api/src/commonMain/kotlin/utils/DeviceInfo.kt
@@ -11,6 +11,7 @@ package net.mamoe.mirai.utils
 
 import io.ktor.utils.io.core.*
 import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.Transient
 import kotlinx.serialization.builtins.serializer
@@ -292,7 +293,7 @@ internal object DeviceInfoManager {
     )
 
     private object DeviceInfoVersionSerializer : KSerializer<DeviceInfo.Version> by SerialData.serializer().map(
-        resultantDescriptor = SerialData.serializer().descriptor.copy("Version"),
+        resultantDescriptor = SerialData.serializer().descriptor,
         deserialize = {
             DeviceInfo.Version(incremental, release, codename, sdk)
         },
@@ -300,6 +301,7 @@ internal object DeviceInfoManager {
             SerialData(incremental, release, codename, sdk)
         }
     ) {
+        @SerialName("Version")
         @Serializable
         private class SerialData(
             val incremental: ByteArray = "5891938".toByteArray(),

--- a/mirai-core-api/src/jvmBaseMain/kotlin/message/data/FileMessage.kt
+++ b/mirai-core-api/src/jvmBaseMain/kotlin/message/data/FileMessage.kt
@@ -120,5 +120,5 @@ public actual interface FileMessage : MessageContent, ConstrainSingle, CodableMe
     }
 
     public actual object Serializer :
-        KSerializer<FileMessage> by FallbackFileMessageSerializer(SERIAL_NAME) // not polymorphic
+        KSerializer<FileMessage> by FallbackFileMessageSerializer()
 }

--- a/mirai-core-api/src/nativeMain/kotlin/message/data/FileMessage.kt
+++ b/mirai-core-api/src/nativeMain/kotlin/message/data/FileMessage.kt
@@ -104,5 +104,5 @@ public actual interface FileMessage : MessageContent, ConstrainSingle, CodableMe
     }
 
     public actual object Serializer :
-        KSerializer<FileMessage> by FallbackFileMessageSerializer(SERIAL_NAME) // not polymorphic
+        KSerializer<FileMessage> by FallbackFileMessageSerializer()
 }

--- a/mirai-core-mock/test/mock/MessageSerializationTest.kt
+++ b/mirai-core-mock/test/mock/MessageSerializationTest.kt
@@ -9,11 +9,10 @@
 
 package net.mamoe.mirai.mock.test.mock
 
-import kotlinx.serialization.KSerializer
+import kotlinx.serialization.*
 import kotlinx.serialization.json.Json
 import kotlinx.serialization.modules.EmptySerializersModule
 import kotlinx.serialization.modules.plus
-import kotlinx.serialization.serializer
 import net.mamoe.mirai.message.MessageSerializers
 import net.mamoe.mirai.message.data.*
 import net.mamoe.mirai.mock.test.MockBotTestBase
@@ -32,7 +31,7 @@ internal class MessageSerializationTest : MockBotTestBase() {
         get() = Json {
             serializersModule = module
             useArrayPolymorphism = false
-            ignoreUnknownKeys = true
+            ignoreUnknownKeys = false
         }
 
     private inline fun <reified T : Any> T.serialize(serializer: KSerializer<T> = module.serializer()): String {
@@ -66,6 +65,13 @@ internal class MessageSerializationTest : MockBotTestBase() {
             deserialized,
             msg
         )
+
+        if (t is SingleMessage) {
+            PolymorphicWrapperContent(t)
+                .serialize(PolymorphicWrapperContent.serializer())
+                .deserialize(PolymorphicWrapperContent.serializer())
+                .let { assert1(t, it.message, msg) }
+        }
     }
 
     private fun assert1(t: Any, deserialized: Any, msg: String) {
@@ -126,4 +132,44 @@ internal class MessageSerializationTest : MockBotTestBase() {
         }
     }
 
+    @Serializable
+    data class PolymorphicWrapperImage(
+        val message: @Polymorphic Image
+    )
+
+    @Serializable
+    data class PolymorphicWrapperContent(
+        val message: @Polymorphic SingleMessage
+    )
+
+    @Test
+    @Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE", "UNRESOLVED_REFERENCE")
+    fun `test serialization for MockImage`() = runTest {
+        val img = this@MessageSerializationTest.bot.uploadMockImage(
+            Image.randomImageContent().toExternalResource().toAutoCloseable()
+        )
+        PolymorphicWrapperImage(img)
+            .serialize(PolymorphicWrapperImage.serializer())
+            .also { println(it) }
+            .deserialize(PolymorphicWrapperImage.serializer())
+    }
+
+    // https://github.com/mamoe/mirai/pull/2414#issuecomment-1386253123
+    @Test
+    fun `test 2414-1386253123`() = runTest {
+        // event
+        val img = this@MessageSerializationTest.bot.uploadMockImage(
+            Image.randomImageContent().toExternalResource().toAutoCloseable()
+        )
+        val msg = buildMessageChain {
+            add("imgUploaded")
+            add(img)
+        }
+
+        val s = format.encodeToString(msg)
+        println(s)
+        println(format.decodeFromString<MessageChain>(s))
+        testSerialization(msg)
+
+    }
 }

--- a/mirai-core/src/commonMain/kotlin/message/data/audio.kt
+++ b/mirai-core/src/commonMain/kotlin/message/data/audio.kt
@@ -163,7 +163,7 @@ internal class OnlineAudioImpl(
     }
 
     object Serializer : KSerializer<OnlineAudioImpl> by Surrogate.serializer().map(
-        resultantDescriptor = Surrogate.serializer().descriptor.copy(OnlineAudio.SERIAL_NAME),
+        resultantDescriptor = Surrogate.serializer().descriptor,
         deserialize = {
             OnlineAudioImpl(
                 filename = filename,
@@ -251,7 +251,7 @@ internal class OfflineAudioImpl(
     }
 
     object Serializer : KSerializer<OfflineAudioImpl> by Surrogate.serializer().map(
-        resultantDescriptor = Surrogate.serializer().descriptor.copy(OfflineAudio.SERIAL_NAME),
+        resultantDescriptor = Surrogate.serializer().descriptor,
         deserialize = {
             OfflineAudioImpl(
                 filename = filename,

--- a/mirai-core/src/commonMain/kotlin/message/protocol/impl/QuoteReplyProtocol.kt
+++ b/mirai-core/src/commonMain/kotlin/message/protocol/impl/QuoteReplyProtocol.kt
@@ -10,6 +10,7 @@
 package net.mamoe.mirai.internal.message.protocol.impl
 
 import net.mamoe.mirai.contact.AnonymousMember
+import net.mamoe.mirai.internal.message.MessageSourceSerializerImpl
 import net.mamoe.mirai.internal.message.protocol.MessageProtocol
 import net.mamoe.mirai.internal.message.protocol.ProcessorCollector
 import net.mamoe.mirai.internal.message.protocol.decode.MessageDecoder
@@ -24,7 +25,6 @@ import net.mamoe.mirai.internal.message.protocol.serialization.MessageSerializer
 import net.mamoe.mirai.internal.message.source.*
 import net.mamoe.mirai.internal.network.protocol.data.proto.ImMsgBody
 import net.mamoe.mirai.message.data.*
-import net.mamoe.mirai.utils.copy
 import net.mamoe.mirai.utils.map
 
 internal class QuoteReplyProtocol : MessageProtocol(PRIORITY_METADATA) {
@@ -100,7 +100,7 @@ internal class QuoteReplyProtocol : MessageProtocol(PRIORITY_METADATA) {
                 MessageSerializer(
                     MessageSource::class,
                     OfflineMessageSourceImplData.serializer().map(
-                        OfflineMessageSourceImplData.serializer().descriptor.copy(MessageSource.SERIAL_NAME),
+                        MessageSourceSerializerImpl.serialDataSerializer().descriptor,
                         { it },
                         {
                             OfflineMessageSourceImplData(

--- a/mirai-core/src/commonTest/kotlin/message/protocol/impl/MarketFaceProtocolTest.kt
+++ b/mirai-core/src/commonTest/kotlin/message/protocol/impl/MarketFaceProtocolTest.kt
@@ -564,6 +564,11 @@ internal class MarketFaceProtocolTest : AbstractMessageProtocolTest() {
         override val message: Dice
     ) : PolymorphicWrapper
 
+    @Serializable
+    data class StaticWrapperRockPaperScissors(
+        override val message: RockPaperScissors
+    ) : PolymorphicWrapper
+
     private fun <M : MarketFace> testPolymorphicInMarketFace(
         data: M,
         expectedSerialName: String,
@@ -585,6 +590,19 @@ internal class MarketFaceProtocolTest : AbstractMessageProtocolTest() {
         testPolymorphicIn(
             polySerializer = StaticWrapperDice.serializer(),
             polyConstructor = ::StaticWrapperDice,
+            data = data,
+            expectedSerialName = null,
+            expectedInstance = expectedInstance,
+        )
+    })
+
+    private fun testStaticRockPaperScissors(
+        data: RockPaperScissors,
+        expectedInstance: RockPaperScissors = data,
+    ) = listOf(dynamicTest("testStaticRockPaperScissors") {
+        testPolymorphicIn(
+            polySerializer = StaticWrapperRockPaperScissors.serializer(),
+            polyConstructor = ::StaticWrapperRockPaperScissors,
             data = data,
             expectedSerialName = null,
             expectedInstance = expectedInstance,
@@ -614,6 +632,22 @@ internal class MarketFaceProtocolTest : AbstractMessageProtocolTest() {
             testPolymorphicInSingleMessage(data, serialName),
             testInsideMessageChain(data, serialName),
             testContextual(data, serialName, targetType = MarketFace::class),
+        )
+    }
+
+    @TestFactory
+    fun `test serialization for RockPaperScissors`(): DynamicTestsResult {
+        val data = RockPaperScissors.PAPER
+
+        val serialName = RockPaperScissors.SERIAL_NAME
+        return runDynamicTests(
+            testPolymorphicInMarketFace(data, serialName),
+            testPolymorphicInMessageContent(data, serialName),
+            testPolymorphicInSingleMessage(data, serialName),
+            testInsideMessageChain(data, serialName),
+            testContextual(data, serialName),
+            testContextual(data, serialName, targetType = MarketFace::class),
+            testStaticRockPaperScissors(data),
         )
     }
 

--- a/mirai-core/src/commonTest/kotlin/message/protocol/impl/QuoteReplyProtocolTest.kt
+++ b/mirai-core/src/commonTest/kotlin/message/protocol/impl/QuoteReplyProtocolTest.kt
@@ -9,8 +9,8 @@
 
 package net.mamoe.mirai.internal.message.protocol.impl
 
+import kotlinx.serialization.Polymorphic
 import kotlinx.serialization.Serializable
-import kotlinx.serialization.json.Json
 import net.mamoe.mirai.internal.message.protocol.MessageProtocol
 import net.mamoe.mirai.internal.message.source.OfflineMessageSourceImplData
 import net.mamoe.mirai.internal.message.toMessageChainOnline
@@ -19,7 +19,6 @@ import net.mamoe.mirai.internal.testFramework.TestFactory
 import net.mamoe.mirai.internal.testFramework.dynamicTest
 import net.mamoe.mirai.internal.testFramework.runDynamicTests
 import net.mamoe.mirai.internal.utils.runCoroutineInPlace
-import net.mamoe.mirai.message.MessageSerializers
 import net.mamoe.mirai.message.data.*
 import net.mamoe.mirai.message.data.MessageSource.Key.quote
 import net.mamoe.mirai.utils.EMPTY_BYTE_ARRAY
@@ -488,18 +487,13 @@ internal class QuoteReplyProtocolTest : AbstractMessageProtocolTest() {
     ///////////////////////////////////////////////////////////////////////////
 
 
-    // TODO: 2022/7/20 MessageSource 在 MessageMetadata 的 scope 多态序列化后会输出 'type' = 'MessageSource', 这是期望的行为.
-    //  但是在反序列化时会错误 unknown field 'type'
-    override val format: Json
-        get() = Json {
-            prettyPrint = true
-            serializersModule = MessageSerializers.serializersModule
-            ignoreUnknownKeys = true
-        }
-
-
     @Serializable
     data class PolymorphicWrapperMessageSource(
+        override val message: @Polymorphic MessageSource
+    ) : PolymorphicWrapper
+
+    @Serializable
+    data class StaticWrapperMessageSource(
         override val message: MessageSource
     ) : PolymorphicWrapper
 
@@ -510,6 +504,19 @@ internal class QuoteReplyProtocolTest : AbstractMessageProtocolTest() {
         testPolymorphicIn(
             polySerializer = PolymorphicWrapperMessageSource.serializer(),
             polyConstructor = ::PolymorphicWrapperMessageSource,
+            data = data,
+            expectedInstance = expectedInstance,
+            expectedSerialName = null,
+        )
+    })
+
+    private fun <M : MessageSource> testStaticInMessageSource(
+        data: M,
+        expectedInstance: M = data,
+    ) = listOf(dynamicTest("testStaticInMessageSource") {
+        testPolymorphicIn(
+            polySerializer = StaticWrapperMessageSource.serializer(),
+            polyConstructor = ::StaticWrapperMessageSource,
             data = data,
             expectedInstance = expectedInstance,
             expectedSerialName = null,
@@ -533,6 +540,7 @@ internal class QuoteReplyProtocolTest : AbstractMessageProtocolTest() {
             testPolymorphicInSingleMessage(data, serialName),
             testInsideMessageChain(data, serialName),
             testContextual(data, serialName),
+            testStaticInMessageSource(data),
         )
     }
 
@@ -548,6 +556,7 @@ internal class QuoteReplyProtocolTest : AbstractMessageProtocolTest() {
             testPolymorphicInSingleMessage(data, serialName, expectedInstance = expected),
             testInsideMessageChain(data, serialName, expectedInstance = expected),
             testContextual(data, serialName, expectedInstance = expected),
+            testStaticInMessageSource(data, expectedInstance = expected),
         )
     }
 }

